File: 무지의 먹방 라이브/code.py
def solution(food_times, k):
    if sum(food_times) <= k:
        return -1  # 모든 음식을 먹는 시간보다 k가 크거나 같으면 더 이상 먹을 음식이 없다.

    # 음식 시간과 원래 인덱스를 함께 저장
    foods = sorted([(time, index) for index, time in enumerate(food_times)])

    total_time = 0  # 총 소요 시간
    previous_time = 0  # 이전 음식까지의 소요 시간
    for i, (time, index) in enumerate(foods):
        # 현재 음식까지 먹는데 필요한 총 시간
        # 여기서 time - previous_time은 남은 음식을 먹는데 필요한 시간
        # len(foods) - i는 남은 음식의 수
        difference = time - previous_time
        if difference * (len(foods) - i) <= k:
            k -= difference * (len(foods) - i)
            total_time += difference
            previous_time = time
        else:
            # 남은 k로 몇 번째 음식을 먹을 수 있는지 찾기
            k %= len(foods) - i
            return sorted(foods[i:], key=lambda x: x[1])[k][1] + 1

# 테스트 케이스
solution([3, 1, 2], 5)

File: 무지의 먹방 라이브/code.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> food_times, long long k) {
    long long total_time = 0;
    for (int time : food_times) {
        total_time += time;
    }

    if (total_time <= k) {
        return -1;
    }

    // 음식 시간과 인덱스를 함께 저장하는 pair 벡터 생성
    vector<pair<int, int>> foods;
    for (int i = 0; i < food_times.size(); i++) {
        foods.push_back({food_times[i], i + 1});
    }

    // 음식을 시간순으로 정렬
    sort(foods.begin(), foods.end());

    total_time = 0;
    long long previous_time = 0;
    long long remaining = foods.size();

    for (int i = 0; i < foods.size(); i++) {
        long long difference = foods[i].first - previous_time;
        if (difference * remaining <= k) {
            k -= difference * remaining;
            total_time += difference;
            previous_time = foods[i].first;
        } else {
            k %= remaining;
            // 인덱스 순으로 다시 정렬하여 k번째 음식 찾기
            sort(foods.begin() + i, foods.end(), [](const pair<int, int>& a, const pair<int, int>& b) {
                return a.second < b.second;
            });
            return foods[i + k].second;
        }
        remaining--;
    }

    return -1;  // 이론적으로 여기에 도달할 수 없음
}

// 예제 사용
int main() {
    vector<int> food_times = {3, 1, 2};
    long long k = 5;
    int result = solution(food_times, k);
    return 0;  // result는 1이 될 것입니다
}